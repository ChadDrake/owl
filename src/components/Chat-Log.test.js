import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './Chat-Log';
import renderer from 'react-test-renderer';

let store = [
  {
    id: 1,
    name: 'Koren Templeton',
    avatar:
      'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true,
  },
  {
    id: 2,
    name: 'Caty Flucker',
    avatar:
      'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: false,
  },
];
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatLog list={store} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
    .create(
      <ChatLog
        id="h"
        type="thumbs up"
        list={store}
        message="heya"
        participantId={1}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
