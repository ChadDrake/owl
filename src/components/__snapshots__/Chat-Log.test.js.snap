// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div>
  Koren Templeton
   
  <img
    alt=""
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  thumbs up
</div>
`;
